uxarray/grid/coordinates.py

```diff
@@ -319,6 +320,91 @@ def _construct_xyz_centroids(node_x, node_y, node_z, face_nodes, n_nodes_per_fac
     return centroids[0, :], centroids[1, :], centroids[2, :]
 
 
+def _populate_edge_centroids(grid, repopulate=False):
+    """Finds the centroids using cartesian averaging of the edges based off the
+    vertices. The centroid is defined as the average of the x, y, z
+    coordinates, normalized.
+
+    Parameters
+    ----------
+    repopulate : bool, optional
+        Bool used to turn on/off repopulating the edge coordinates of the centroids
+    """
+
+    node_x = grid.node_x.values
+    node_y = grid.node_y.values
+    node_z = grid.node_z.values
+    edge_nodes_con = grid.edge_node_connectivity.values
+
+    if "edge_lon" not in grid._ds or repopulate:
+        # Construct the centroids if there are none stored
+        if "edge_x" not in grid._ds:
+            centroid_x, centroid_y, centroid_z = _construct_edge_centroids(
+                node_x, node_y, node_z, edge_nodes_con
+            )
+
+        else:
+            # If there are cartesian centroids already use those instead
+            centroid_x, centroid_y, centroid_z = grid.edge_x, grid.edge_y, grid.edge_z
+
+        # Convert from xyz to latlon
+        centroid_lon, centroid_lat = _get_lonlat_from_xyz(
+            centroid_x, centroid_y, centroid_z
+        )
+    else:
+        # Convert to xyz if there are latlon centroids already stored
+        centroid_lon, centroid_lat = grid.edge_lon.values, grid.edge_lat.values
+        centroid_x, centroid_y, centroid_z = _get_xyz_from_lonlat(
+            centroid_lon, centroid_lat
+        )
+
+    # Populate the centroids
+    if "edge_lon" not in grid._ds or repopulate:
+        grid._ds["edge_lon"] = xr.DataArray(
+            centroid_lon, dims=["n_edge"], attrs={"units": "degrees_east"}
+        )
+        grid._ds["edge_lat"] = xr.DataArray(
+            centroid_lat, dims=["n_edge"], attrs={"units": "degrees_north"}
+        )
+
+    if "edge_x" not in grid._ds or repopulate:
+        grid._ds["edge_x"] = xr.DataArray(
+            centroid_x, dims=["n_edge"], attrs={"units": "meters"}
+        )
+
+        grid._ds["edge_y"] = xr.DataArray(
+            centroid_y, dims=["n_edge"], attrs={"units": "meters"}
+        )
+
+        grid._ds["edge_z"] = xr.DataArray(
+            centroid_z, dims=["n_edge"], attrs={"units": "meters"}
+        )
+
+
+def _construct_edge_centroids(node_x, node_y, node_z, edge_nodes_con):
+    """Constructs the xyz centroid coordinate for each edge using Cartesian
+    Averaging."""
+    centroids = np.zeros((3, edge_nodes_con.shape[0]), dtype=np.float64)
```

Comment:
```suggestion
    centroids = np.zeros((3, edge_nodes_con.shape[0]), dtype=node_x.dtype)
```

Minor change, but I suggest using the same data type as the node coordinates.

Reply:
If `node_x` is int32 it seems to cause an error in `node_xyz_to_lonlat_rad` due to numba, this happens in the triangle test case as it is just integer points. So using `np.float64` seems the better route, as it forces the centroids to be compatible with the conversion function.

Reply:
Interesting, is there any cases where `node_x` is an integer type? 

This is fine with me though, this works.